ui/src/organizations/components/TaskExportOverlay.tsx

```diff
@@ -1,60 +1,86 @@
 import React, {PureComponent} from 'react'
 import {withRouter, WithRouterProps} from 'react-router'
+import {connect} from 'react-redux'
 
 // Components
 import ExportOverlay from 'src/shared/components/ExportOverlay'
 
-// Utils
-import {taskToTemplate} from 'src/shared/utils/resourceToTemplate'
+// Actions
+import {convertToTemplate as convertToTemplateAction} from 'src/tasks/actions'
+import {clearExportTemplate as clearExportTemplateAction} from 'src/templates/actions'
 
-// APIs
-import {client} from 'src/utils/api'
+// Types
+import {AppState} from 'src/types/v2'
 import {DocumentCreate} from '@influxdata/influx'
+import {RemoteDataState} from 'src/types'
 
-interface State {
-  taskTemplate: DocumentCreate
-  orgID: string
+interface OwnProps {
+  params: {id: string; orgID: string}
+}
+
+interface DispatchProps {
+  convertToTemplate: typeof convertToTemplateAction
+  clearExportTemplate: typeof clearExportTemplateAction
 }
 
-interface Props extends WithRouterProps {
-  params: {id: string}
+interface StateProps {
+  taskTemplate: DocumentCreate
+  status: RemoteDataState
+  orgID: string
 }
 
-class TaskExportOverlay extends PureComponent<Props, State> {
-  public state: State = {taskTemplate: null, orgID: null}
+type Props = OwnProps & StateProps & DispatchProps & WithRouterProps
 
+class TaskExportOverlay extends PureComponent<Props> {
   public async componentDidMount() {
     const {
       params: {id},
+      convertToTemplate,
     } = this.props
 
-    const task = await client.tasks.get(id)
-    const taskTemplate = taskToTemplate(task)
-
-    this.setState({taskTemplate, orgID: task.orgID})
+    convertToTemplate(id)
   }
 
   public render() {
-    const {taskTemplate, orgID} = this.state
-    if (!taskTemplate) {
-      return null
-    }
+    const {taskTemplate, status} = this.props
 
     return (
       <ExportOverlay
         resourceName="Task"
         resource={taskTemplate}
         onDismissOverlay={this.onDismiss}
-        orgID={orgID}
+        orgID={this.orgID}
+        status={status}
       />
     )
   }
 
+  private get orgID() {
+    const orgFromExistingResource = this.props.orgID
+    const orgInRoutes = this.props.params.orgID
+    return orgFromExistingResource || orgInRoutes
+  }
+
   private onDismiss = () => {
-    const {router} = this.props
+    const {router, clearExportTemplate} = this.props
 
     router.goBack()
```

Comment:
one thing I've seen with `goBack` is if signout occurs because of session expiration, and then they log back in and exit the `TaskExportOverlay` they'll be sent back to the `/signin?returnTo=/tasks/:id/export` page by `router.goBack()`. 

See https://github.com/influxdata/influxdb/issues/12816

Reply:
There are other problems with goback() as well :(